src/components/track-tile/TrackTile.tsx

```diff
@@ -1,106 +1,215 @@
+  saveTrack,
+  undoRepostTrack,
+  unsaveTrack
+} from 'audius-client/src/common/store/social/tracks/actions'
+import {
+  OverflowAction,
+  OverflowSource
+} from 'audius-client/src/common/store/ui/mobile-overflow-menu/types'
+import { requestOpen as requestOpenShareModal } from 'audius-client/src/common/store/ui/share-modal/slice'
+import { open as openOverflowMenu } from 'common/store/ui/mobile-overflow-menu/slice'
+import { isEqual } from 'lodash'
+import {
+  Animated,
+  Easing,
+  GestureResponderEvent,
+  Pressable,
+  StyleSheet
+} from 'react-native'
+import { useDispatch, useSelector } from 'react-redux'
 
+import { BaseStackParamList } from 'app/components/app-navigator/types'
 import { TrackTileProps } from 'app/components/track-tile/types'
+import { useDispatchWeb } from 'app/hooks/useDispatchWeb'
+// import { usePushRouteWeb } from 'app/hooks/usePushRouteWeb'
+import { useSelectorWeb } from 'app/hooks/useSelectorWeb'
 import { useThemedStyles } from 'app/hooks/useThemedStyles'
-import { flexCol, flexRow } from 'app/styles'
+import { getPlaying, getPlayingUid } from 'app/store/audio/selectors'
 import { ThemeColors } from 'app/utils/theme'
 
 import { TrackBannerIcon, TrackBannerIconType } from './TrackBannerIcon'
 import { TrackTileBottomButtons } from './TrackTileBottomButtons'
 import { TrackTileCoSign } from './TrackTileCoSign'
+import { TrackTileContainer } from './TrackTileContainer'
 import { TrackTileMetadata } from './TrackTileMetadata'
 import { TrackTileStats } from './TrackTileStats'
 import { TrackTileTopRight } from './TrackTileTopRight'
 
 const createStyles = (themeColors: ThemeColors) =>
   StyleSheet.create({
-    container: {
-      ...flexRow(),
-      minHeight: 152,
-      borderColor: themeColors.neutralLight8,
-      backgroundColor: themeColors.white,
-      borderWidth: 1,
-      borderRadius: 8,
-      maxWidth: 400,
-      marginHorizontal: 'auto',
-      marginBottom: 12,
-      elevation: 3,
-      shadowOpacity: 0.15,
-      shadowOffset: { width: 0, height: 1 },
-      shadowRadius: 3
-    },
     mainContent: {
-      ...flexCol(),
       flex: 1
     }
   })
 
-export const TrackTile = ({
-  artistName,
-  coSign,
-  coverArtSizes,
-  duration,
-  fieldVisibility,
-  goToTrackPage,
-  goToArtistPage,
-  hasCurrentUserReposted,
-  hasCurrentUserSaved,
-  id,
+export const TrackTile = (props: TrackTileProps) => {
+  const { uid } = props
+
+  // Using isEqual as the equality function to prevent rerenders due to object references
+  // not being preserved when syncing redux state from client.
+  // This can be removed when no longer dependent on web client
+  const track = useSelectorWeb(state => getTrack(state, { uid }), isEqual)
+  const user = useSelectorWeb(state => getUserFromTrack(state, { uid }))
+  if (!track || !user) {
+    console.warn('Track or user missing for TrackTile, preventing render')
+    return null
+  }
+
+  return <TrackTileComponent {...props} track={track} user={user} />
```

Comment:
is the main reason to split these up so we can be sure track and user are there before including the other hooks in TrackTileComponent?

Reply:
This is kind of a "null" guard. @piazzatron built an awesome pattern for this in audius-client, that maybe we should adopt (I ran into similar w/ play bar).

https://github.com/AudiusProject/audius-client/blob/5b0b7683d6fda5148222659d5014356afefb26ec/src/pages/remixes-page/components/desktop/RemixesPage.tsx#L37

The concept is that it captures the nulls so you don't have type issues downstream

Reply:
word yeah thats what it seemed like, makes sense, thanks!

Reply:
Yeah I was thinking through using withNullGuard, but with redux hooks we need a separate component anyway. And in that case, withNullGuard essentially just does a null check, which we could easily do in the component like I'm doing here

Maybe this is a place where we don't use redux hooks and opt for `connect` instead?

Reply:
i like what you did, personally its easier to read than connect imo